### PR TITLE
Make nifikop OpenShift compatible

### PR DESCRIPTION
**Pull request by GitarPlayer**
| Q               | A
| --------------- | ---
| Bug fix?        | no|yes
| New feature?    | yes|yes
| API breaks?     | no|yes
| Deprecations?   | no|yes
| Related tickets | mentioned in #250 
| License         | Apache 2.0

### What's in this PR?
1. I updated the kubebuilder RBAC annotations so the operator works on OpenShift.
2. I added a new variable to install the helm operator while specifying the RunAsUser
3. I added a nificlusters.nifi.konpyutaika.com sample for OpenShift 

### Why?
As it is NifiKop does not run on OpenShift without custom day 2 modifications

### Additional context
I checked the install on OpenShift
```bash
# Tried on a clean AKS OpenShift cluster
oc version  
Client Version: 4.12.9
Kustomize Version: v4.5.7
Server Version: 4.10.54
Kubernetes Version: v1.23.12+8a6bfe4
# Create namespaces for Zookeeper and NiFi
oc create ns zookeeper
oc create ns nifi

# Install the CustomResourceDefinitions and cert-manager itself
kubectl apply -f https://github.com/jetstack/cert-manager/releases/download/v1.7.2/cert-manager.yaml

# Get UID range for the operator install from the nifi namespace
uid=$(kubectl get namespace nifi -o=jsonpath='{.metadata.annotations.openshift\.io/sa\.scc\.supplemental-groups}' | sed 's/\/10000$//' | tr -d '[:space:]')

# Install Nifi operator using helm
helm install nifikop \
    nifikop \
    --namespace=nifi \
    --version 1.1.1 \
    --set image.tag=v1.1.1-release \
    --set resources.requests.memory=256Mi \
    --set resources.requests.cpu=250m \
    --set resources.limits.memory=256Mi \
    --set resources.limits.cpu=250m \
    --set namespaces={"nifi"} \
    --set runAsUser=$uid

# Get UID range for the Zookeeper operator from the zookeeper namespace
zookeper_uid=$(kubectl get namespace zookeeper -o=jsonpath='{.metadata.annotations.openshift\.io/sa\.scc\.supplemental-groups}' | sed 's/\/10000$//' | tr -d '[:space:]')

# Get the default storage class for the cluster
sc=$(kubectl get storageclass -o=jsonpath='{range .items[?(@.metadata.annotations.storageclass\.kubernetes\.io/is-default-class=="true")]}{.metadata.name}{end}')

# Install Zookeeper using helm
helm install zookeeper bitnami/zookeeper \
    --set resources.requests.memory=256Mi \
    --set resources.requests.cpu=250m \
    --set resources.limits.memory=256Mi \
    --set resources.limits.cpu=250m \
    --set global.storageClass=$sc \
    --set networkPolicy.enabled=true \
    --set replicaCount=3 \
    --set containerSecurityContext.runAsUser=$zookeper_uid \
    --set podSecurityContext.fsGroup=$zookeper_uid \
    --namespace zookeeper

# Use the UID for the NiFi operator to set the fsGroup and runAsUser
sed -i "s/1000690000/$uid/g" config/samples/openshift

# Use the default storage class for the cluster to set the persistent volume claim
sed -i "s/standard/$sc/g" config/samples/openshift

# Apply the configuration for the NiFi operator
oc apply -f config/samples/openshift -n nifi

# Expose the NiFi service as a route
oc expose svc -n nifi simplenifi-headless

# Get the route for the NiFi service
route=$(kubectl get route simplenifi-headless -n nifi -o=jsonpath='{.spec.host}')

# Open the NiFi UI in Firefox using the route
firefox http://$route/nifi

```

### Checklist

- [x] Implementation tested
- [x] User guide and development docs updated (if needed) 
- [x] Append changelog with changes

### To Do
- [x] User guide and development docs updated (if needed)  where can I do this? This DOC https://konpyutaika.github.io/nifikop/docs/2_deploy_nifikop/1_quick_start sources I did not find. I will update the changelog once I updated the DOC.

### CODE REVIEW

File: helm/nifikop/values.yaml [hunk 1]
```diff
@@ -33,6 +33,9 @@ resources:
 ## Ref: https://kubernetes.io/docs/tasks/network/customize-hosts-file-for-pods/
 hostAliases: []
 
+## RunAsUser for OpenShift compatibility
+runAsUser: 1001
```
Comment (r65535) [inline helm/nifikop/values.yaml:37]: The default should probably stay as `1000` to keep functionality the same?

**Comment by r65535** [comment]
Looks good to me!

It might be worth adding a page to the docs for running under openshift? Or adding a section to an existing page with the extra config required 😄

File: helm/nifikop/values.yaml (hunk 1 above)
Comment (GitarPlayer) [reply helm/nifikop/values.yaml]: totally agree my bad 1001 is an OpenShift convention uid.

**Comment by GitarPlayer** [comment]
> Looks good to me!
> 
> It might be worth adding a page to the docs for running under openshift? Or adding a section to an existing page with the extra config required 😄
Of course where can I edit the documentation. I did a grep -Ri 'The NiFiKop operator uses cert-manager f' in this repo and I did not find anything. Where is this doc hosted: https://konpyutaika.github.io/nifikop/?

**Comment by mh013370** [comment]
I pulled this branch and built it just to make sure any kubebuilder-generated files get updated. I did find one that was missed (i.e. `config/rbac/role.yaml`). Can you please run `make build` locally and make sure that file get committed?

**Comment by GitarPlayer** [comment]
Sorry I was out of office for a week. My bad I added the ```make build``` changes.

**Review by mh013370 — APPROVED** [review]

**Comment by mh013370** [comment]
@GitarPlayer : this PR needs its commits all signed and i can merge. Thanks!

**Comment by juldrixx** [comment]
Why did this PR have 545 commits and old ones in it?

**Comment by GitarPlayer** [comment]
> @GitarPlayer : this PR needs its commits all signed and i can merge. Thanks!

Can you point me out how to sign all my previous commits without making a git mess?

> Why did this PR have 545 commits and old ones in it?

 christian_bucheli@hotmail.com
I tried signing my old commits but I ended up creating a mess. I am not sure what the correct procedure is to sign all my old commits.

**Comment by juldrixx** [comment]
> > @GitarPlayer : this PR needs its commits all signed and i can merge. Thanks!
> 
> Can you point me out how to sign all my previous commits without making a git mess?
> 
> > Why did this PR have 545 commits and old ones in it?
> 
> [christian_bucheli@hotmail.com](mailto:christian_bucheli@hotmail.com) I tried signing my old commits but I ended up creating a mess. I am not sure what the correct procedure is to sign all my old commits.

The issue is that you have a bunch of old commits the PR that are triggering an alert because of some secrets that aren't secrets.

**Comment by GitarPlayer** [comment]
> > > @GitarPlayer : this PR needs its commits all signed and i can merge. Thanks!
> > 
> > 
> > Can you point me out how to sign all my previous commits without making a git mess?
> > > Why did this PR have 545 commits and old ones in it?
> > 
> > 
> > [christian_bucheli@hotmail.com](mailto:christian_bucheli@hotmail.com) I tried signing my old commits but I ended up creating a mess. I am not sure what the correct procedure is to sign all my old commits.
> 
> The issue is that you have a bunch of old commits the PR that are triggering an alert because of some secrets that aren't secrets.

This is related to the last three commits where I tried to sign all my commits. If I revert them / remove them from the history, this should be fixed. But do you know the easiest way to sign old commits done by me?

**Comment by indiealexh** [comment]
I'd love access to this compatibility feature this week, is there anything I can do to help this move forward?

**Comment by juldrixx** [comment]
> I'd love access to this compatibility feature this week, is there anything I can do to help this move forward?

Fix the PR, as it is broken (500+ commits).

**Comment by GitarPlayer** [comment]
> I'd love access to this compatibility feature this week, is there anything I can do to help this move forward?

I will simply create a new fork and create a new PR and sign all my commits. This is easier than fixing this mess I've created.

**Pull request closed without merging**